Snabble/Core/Payment/PaymentMethodDetails.swift

```diff
@@ -115,6 +118,8 @@ struct TegutEmployeeData: Codable, EncryptedPaymentData, Equatable {
         self.serial = try container.decode(String.self, forKey: .serial)
         self.displayName = try container.decode(String.self, forKey: .displayName)
         self.cardNumber = (try? container.decodeIfPresent(String.self, forKey: .cardNumber)) ?? ""
+        let projectId = try container.decodeIfPresent(Identifier<Project>.self, forKey: .projectId)
```

Comment:
Wenn keiner da ist, sollten wir dann nicht mit einem Optional `Identifier<Project>` arbeiten? Es erhöht die Verständlichkeit wenn mit Optionals gearbeitet wird, da `empty` hier ein potentiell validier wert sein könnte.

Reply:
Das führt aber nur zu dauernden unwraps oder checks, obwohl eigentlich klar ist dass das Ding immer einen Wert haben muss. `""` ist auch garantiert nie eine valide Projekt-ID.

Reply:
Wenn es immer da ist, warum werden wir dann nicht `decode(_:forKey:)`?

Reply:
Weil es in den alten Records nicht da ist, und wenn die da sind darf das Parsing nicht scheitern.

Reply:
Sehe ich es richtig, das im `PaymentMethodDetail` das Element auch ein Optional ist? Es gibt wenn ich es richtig sehe keinen direkt Aufruf, sondern nur über `PaymentMethodDetail`, dadurch sollte es doch kein Mehraufwand sein.

Zusätzlich: Ist es für dich ein Unterschied ob ich prüfen den Project Identifier auf Leer prüfen oder ob er vorhanden ist? 🤔

Reply:
Der Unterschied ist in der Semantik der Daten. Es kann ab jetzt keine Kreditkartendaten (mehr) geben, die keine ProjektID haben, deswegen ist das Ding da nicht Optional, und ich will nicht suggerieren dass das evtl doch so wäre. Andersrum hat aber nicht alles was ein PaymentMethodDetail hat so eine ID, also muss es auf der Ebene optional sein.  Diese ganze Mimik ist nur dazu da uns die Migration, also das sofortige Löschen dieser alten Daten ohne ID zu ermöglichen.

Reply:
> Der Unterschied ist in der Semantik der Daten. Es kann ab jetzt keine Kreditkartendaten (mehr) geben, die keine ProjektID haben, deswegen ist das Ding da nicht Optional, und ich will nicht suggerieren dass das evtl doch so wäre.

Sehr guter Punkt 👍 Ist es uns möglich die Migration der Altdaten auf Neue mit ProjectID zu ziehen? Ansonsten behalten wir uns auf ewig die Krücke das wir potentiell auf `empty` ProjectID prüfen müssten.